pilot/pkg/serviceregistry/mcp/controller.go

```diff
@@ -329,3 +346,46 @@ func extractNameNamespace(metadataName string) (string, string) {
 	}
 	return "", segments[0]
 }
+
+func (c *controller) removeConfig(kind resource.GroupVersionKind, configName []string) {
+	if len(configName) == 0 {
+		return
+	}
```

Comment:
You can omit this check. There is nothing after the for loop below and it will be skipped if there are no resources to remove.

Reply:
Thank you very much for your suggestion. done